src/preferenceCenter/preferenceCenter.domcomponents.js

```diff
@@ -0,0 +1,239 @@
+              this.handleChecked
+            );
+          },
+          handleChecked: function () {
+            this.el.checked = !!this.model.get('attributes').checked;
+          },
+        },
+      }),
+      dc.addType('label', {
+        extend: 'text',
+        isComponent: function (e) {
+          return 'LABEL' == e.tagName;
+        },
+        model: { defaults: { tagName: 'label', components: 'Label' } },
+      });
+    dc.addType('option', {
+      isComponent: (el) => el.tagName == 'OPTION',
+
+      model: {
+        defaults: {
+          tagName: 'option',
+          layerable: false,
+          droppable: false,
+          draggable: false,
+          highlightable: false,
+        },
+      },
+    });
+
+    const createOption = (value, name) => ({
+      type: 'option',
+      components: name,
+      attributes: { value },
+    });
+
+    dc.addType('select', {
+      extend: 'input',
+      isComponent: (el) => el.tagName == 'SELECT',
+
+      model: {
+        defaults: {
+          tagName: 'select',
+          components: [
+            createOption('email', 'Email'),
+            //createOption('opt2', 'Option 2'),
+          ],
+          traits: [
+            nameTrait,
+            {
+              name: 'options',
+              type: 'select-options',
+            },
+            requiredTrait,
+          ],
+        },
+      },
+
+      view: {
+        events: {
+          mousedown: (e) => e.preventDefault(),
+        },
+      },
+    });
+
+    const domc = this.editor.DomComponents;
+    const defaultType = domc.getType('default');
+    const defaultModel = defaultType.model;
+    const defaultView = defaultType.view;
+
+    dc.addType('date', {
+      model: defaultModel.extend(
+        {
+          defaults: {
+            ...defaultModel.prototype.defaults,
+            'custom-name': 'date',
+            tagName: 'input',
+            type: 'date',
+            attributes: {
+              'data-gjs-type': 'date',
+              'data-slot': 'date',
+            },
+            droppable: false, // Can't drop other elements inside
+            copyable: false, // Do not allow to duplicate the component
+            script: function () {
+              var input = this;
+              var initDateRange = function () {
+                var input = this;
+                const options = {
+                  singleDatePicker: true,
+                  showDropdowns: true,
+                };
+                $(input).daterangepicker(options);
+              };
+
+              if (
+                typeof daterangepicker == 'undefined' ||
+                typeof jQuery == 'undefined' ||
+                typeof moment == 'undefined'
+              ) {
+                if (typeof jQuery == 'undefined') {
+                  var jquery = document.createElement('script');
+                  jquery.src =
+                    '//cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js';
```

Comment:
I think jquery is already loaded with Mautic. Do we really need to load it again?

Reply:
It's a component specific requirement as to load datepicker into the canvas we will need it. I think the jquery loaded by mautic is general for all pages of the project. But inside the builder if its already loaded then it will not give undefined and rest of the code will work.

Reply:
@adiux the issue with this code is when we drag date element first time it's get loaded properly date input works fine. But if we delete the element and drag it again to canvas then it does not open datepicker. If you can look into the error that will be really helpful.  The name of the block which contains the date component is Channel Frequency. Thanks